src-java/flowhs-topology/flowhs-storm-topology/src/main/java/org/openkilda/wfm/topology/flowhs/bolts/YFlowPathSwapHubBolt.java

```diff
@@ -0,0 +1,246 @@
+        currentKey = pullKey(input);
+        YFlowPathSwapRequest payload = pullValue(input, FIELD_ID_PAYLOAD, YFlowPathSwapRequest.class);
+        try {
+            yFlowPathSwapService.handleRequest(currentKey, pullContext(input), payload);
+        } catch (DuplicateKeyException e) {
+            log.error("Failed to handle a request with key {}. {}", currentKey, e.getMessage());
+        }
+    }
+
+    @Override
+    protected void onWorkerResponse(Tuple input) throws PipelineException {
+        String operationKey = pullKey(input);
+        currentKey = KeyProvider.getParentKey(operationKey);
+        SpeakerResponse speakerResponse = pullValue(input, FIELD_ID_PAYLOAD, SpeakerResponse.class);
+        try {
+            yFlowPathSwapService.handleAsyncResponse(currentKey, speakerResponse);
+        } catch (UnknownKeyException e) {
+            log.error("Received a response with unknown key {}.", currentKey);
+        }
+    }
+
+    @Override
+    public void onTimeout(String key, Tuple tuple) {
+        currentKey = key;
+        try {
+            yFlowPathSwapService.handleTimeout(key);
+        } catch (UnknownKeyException e) {
+            log.error("Failed to handle a timeout event for unknown key {}.", currentKey);
+        }
+    }
+
+    @Override
+    public void sendSpeakerRequest(@NonNull SpeakerRequest command) {
+        String commandKey = KeyProvider.joinKeys(command.getCommandId().toString(), currentKey);
+
+        Values values = new Values(commandKey, command);
+        emitWithContext(HUB_TO_SPEAKER_WORKER.name(), getCurrentTuple(), values);
+    }
+
+    @Override
+    public void sendNorthboundResponse(@NonNull Message message) {
+        emitWithContext(Stream.HUB_TO_NB_RESPONSE_SENDER.name(), getCurrentTuple(), new Values(currentKey, message));
+    }
+
+    @Override
+    public void sendHistoryUpdate(@NonNull FlowHistoryHolder historyHolder) {
+        InfoMessage message = new InfoMessage(historyHolder, getCommandContext().getCreateTime(),
+                getCommandContext().getCorrelationId());
+        emitWithContext(Stream.HUB_TO_HISTORY_TOPOLOGY_SENDER.name(), getCurrentTuple(),
+                new Values(historyHolder.getTaskId(), message));
+    }
+
+    @Override
+    public void cancelTimeoutCallback(String key) {
+        cancelCallback(key);
+    }
+
+    @Override
+    public void sendInactive() {
+        getOutput().emit(ZkStreams.ZK.toString(), new Values(deferredShutdownEvent, getCommandContext()));
+        deferredShutdownEvent = null;
+    }
+
+    @Override
+    public void sendPeriodicPingNotification(String flowId, boolean enabled) {
+        PeriodicPingCommand payload = new PeriodicPingCommand(flowId, enabled);
+        Message message = new CommandMessage(payload, getCommandContext().getCreateTime(),
+                getCommandContext().getCorrelationId());
+        emitWithContext(Stream.HUB_TO_PING_SENDER.name(), getCurrentTuple(), new Values(currentKey, message));
+    }
```

Comment:
Because FlowPathSwapHubBolt implements it and FlowPathSwapFsm uses it. YFlowPathSwapHub in its turn utilizes FlowPathSwapFsm for sub-flows, so the method has to be provided.